packages/components/pie-chip/src/index.ts

```diff
@@ -97,31 +110,33 @@ export class PieChip extends LitElement implements ChipProps {
             isLoading,
             isDismissible,
         } = this;
+        const showCloseButton = isDismissible && isSelected;
 
         const classes = {
             'c-chip': true,
             [`c-chip--${variant}`]: true,
             'c-chip--selected': isSelected,
-            'c-chip--dismissible': isDismissible,
+            'c-chip--dismissible': showCloseButton,
+            'c-chip--disabled': disabled,
```

Comment:
NIT – as this is a state that would likely be updated via JS, I'd make this `is-disabled`.

Reply:
the current convention in the codebase is to use the component name postfixed with the state at the moment in the whole codebase 👀 do you think it's worth a convention discussion and see if we wanna refactor to that?

Reply:
I'd say if we want to standardise this, let's have a separate discussion and address in a different PR 👍. You should probably argue the same for `selected`. Worth doing a full audit of our components before making this change IMO

Reply:
Yeh, I noticed that selected should also be like this too, but as Raouf hadn't touched that, didn't bring it up :)

The main guide I work to is:
- If it's something that is intended as a single use state via prop (like variant) then is should be a component modifer `c-chip--{variant}`
- If it's something that is likely to be modified by JS, such as active states, loading or disabled, it should use a state class `is-`

Reply:
@ashleynolan forgive me but I can only see `--disabled` used in the codebase (not sure if I'm searching incorrectly):

<img width="324" alt="Screenshot 2024-10-16 at 14 51 01" src="https://github.com/user-attachments/assets/4ced26ca-5db2-4c58-b865-acd4ea1a98eb">

<img width="332" alt="Screenshot 2024-10-16 at 14 51 17" src="https://github.com/user-attachments/assets/547289e0-df56-4912-af9b-1bcdacb5e932">

Can we not address this separately? I'd prefer we agree on the convention and update all of them under a single ticket. Currently what Raouf as written would technically be the consistent approach 😄

Reply:
As long as it happens, I don't mind – but yeh, the instances of card/checkbox/radio should change to be `is-disabled`, as that matches up with the naming like `is-loading` (which is also in `pie-chip` and `pie-button`)

Reply:
i created a follow-up ticket and will refine it tomorrow 🤝